src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/transforms/columns/utils.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { DataType, TimeScaleUnit } from '@kbn/visualizations-plugin/common';
+import type { Query } from '@kbn/es-query';
+import { fromFilterAPIToLensState, fromFilterLensStateToAPI } from './filter';
+import type {
+  LensApiAllMetricOperations,
+  LensApiReferableMetricOperations,
+} from '../../schema/metric_ops';
+import type {
+  AnyLensStateColumn,
+  AnyMetricLensStateColumn,
+  ReferableMetricLensStateColumn,
+} from './types';
+import type { LensApiAllOperations } from '../../schema';
+
+export const LENS_EMPTY_AS_NULL_DEFAULT_VALUE = false;
+
+export function getLensStateMetricSharedProps(
+  options: {
+    time_scale?: TimeScaleUnit;
+    reduced_time_range?: string;
+    time_shift?: string;
+    filter?: { query: string; language: 'kuery' | 'lucene' };
+    label?: string;
+  },
+  defaultLabel: string = '',
+  dataType: DataType = 'number'
+) {
+  return {
+    dataType,
+    isBucketed: false,
+    filter: fromFilterAPIToLensState(options.filter),
+    timeScale: options.time_scale,
+    reducedTimeRange: options.reduced_time_range,
+    timeShift: options.time_shift,
+    label: options.label ?? defaultLabel,
```

Comment:
defaultLabel is not needed as we set customLabel to false if options.label is not provided

Reply:
i think this falls under the group of options/defaults where we shouldnt be trying to fill them in as lens will provide smarter defaults based on the actual data.

Reply:
still relevant

Reply:
`label` type is mandatory in Lens.
We have to deal with that until the `label` prop is made optional on the Lens state side.

Reply:
thats fine, but we should always set it to '' if input label is not set, or to some other constant string, we shouldnt need to complicate transformations with setting some default value